lib/stat_tracker.rb

```diff
@@ -93,24 +93,20 @@ def avg_goals_by_team
     teams_data.each do |team_id, game_info|
       sum = game_info.sum { |game| game.goals}
       num_games = game_info.count
-      teams_data[team_id] = (sum / num_games)
+      teams_data[team_id] = (sum / num_games.to_f).round(2)
     end
   end
 
   def best_offense
-    highest = avg_goals_by_team.max_by do |team_id, avg_goals|
-      avg_goals
-    end
-    team = @teams.find { |team| team.team_id == highest[0]}
-    team.team_name
+    highest = avg_goals_by_team.max_by { |team_id, avg_goals|  avg_goals }
+    best = @teams.find { |team| team.team_id == highest[0]}
+    best.team_name
   end
 
   def worst_offense
-    lowest = avg_goals_by_team.min_by do |team_id, avg_goals|
-      avg_goals
-    end
-    team = @teams.find { |team| team.team_id == lowest[0]}
-    team.team_name
+    lowest = avg_goals_by_team.min_by { |team_id, avg_goals|avg_goals }
+    worst = @teams.find { |team| team.team_id == lowest[0]}
+    worst.team_name
```

Comment:
I like the variable renaming to better reflect the data.

Reply:
Good stuff! This looks clean.